gtsam/discrete/AlgebraicDecisionTree.h

```diff
@@ -36,7 +36,7 @@ namespace gtsam {
    * @ingroup discrete
    */
   template <typename L>
-  class GTSAM_EXPORT AlgebraicDecisionTree : public DecisionTree<L, double> {
+  class AlgebraicDecisionTree : public DecisionTree<L, double> {
```

Comment:
I'm a bit confused by this change. Don't the gtsam.discrete tests already pass?

Reply:
even if the tests are passing, it shouldn't be added this export, it will be create of linkage error when using it in different combination.

Reply:
I see. Can you please give an example so I can better understand this? That way we can come up with a good solution together.

Reply:
I don't have an example for `AlgebraicDecisionTree` class. But you can see other template class here that I remove the export, bring them back and see the linkage error they will bring (they will usually bring errors on different classes that why it hard to find it).
So the thumb rule here: for template class it shouldn't be an export. (Also you can see it work without it). and regular class if use it, it need an export.
I hope it more clear.
Any other comments overall?